### PR TITLE
Add mycroft-config tool to manage config files

### PR DESCRIPTION
**Pull request by penrods**
The new mycroft-config utility simplifies management of the the
various configuration files that control Mycroft.  Commands include:
edit, show, get, set, reload.

* ```mycroft-config edit (default|remote|system|user)```
Open the requested file in an editor (nano by default), performing
JSON validation warnings to minimize accidental edit errors.

* ```mycroft-config show (default|remote|system|user)```
Dump the selected file with nice highlighting using jq

* ```mycroft-config get <var>```
Load the mycroft.conf stack, outputting the effective value(s).  The
<var> is specified using jq-style specifiers.  For example:
   mycroft-config get enclosure.platform
   mycroft-config get location
   mycroft-config get

* ```mycroft-config set <var> <value>```
Set the given variable in the "user" configuration file.  Examples:
   mycroft-config set lang "es-es"
   mycroft-config set tts.mimic.voice "ap"

* ```mycroft-config reload```
Sends a signal on the messagebus to tell services to reload configs.
This automatically occurs after a 'set' or 'edit' command.

### CODE REVIEW

**Comment by JarbasAl** [comment]
this should reduce support requests by 70%!

most of the time "mycroft is broken" it's a bad comma messing json, probably in core config file....

File: bin/mycroft-config [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    echo "  ${script} set test.subvalue \"foo\" "
+
+    exit 1
+}
+
+################################################################
+# Setup stuff based on the environment
+
+VIEWER="nano --syntax=json --view"
+if [ -z "$EDITOR" ] ; then
```
Comment (forslund) [inline bin/mycroft-config]: should the sensible-editor link be checked as an alternative?

File: bin/mycroft-config [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        *)
+            echo "ERROR: Unknown name '${1}'."
+            echo "       Must be one of: default, remote, system, or user"
+            exit 1
+    esac
+}
+
+################################################################
+
+function edit-config() {
```
Comment (forslund) [inline bin/mycroft-config]: Let's use underscores instead of hyphens to keep constency and portability

Comment (penrods) [reply bin/mycroft-config]: Done

File: bin/mycroft-config (hunk 1 above)
Comment (penrods) [reply bin/mycroft-config]: Done

**Comment by forslund** [comment]
Rebased and added a warning for when opening in view mode instead of the expected edit mode. Merging now.

**Pull request merged**